core/src/util/usUtils.cpp

```diff
@@ -388,46 +388,43 @@ std::string GetLastErrorStr()
 #endif
 }
 
-static MsgHandler handler = 0;
-
-MsgHandler installMsgHandler(MsgHandler h)
-{
-  MsgHandler old = handler;
-  handler = h;
-  return old;
-}
-
-void message_output(MsgType msgType, const char *buf)
+void TerminateForDebug(const std::exception_ptr ex)
```

Comment:
What if we create a new macro (```DIAG_LOG_TERMINATE(message)```?) that acts like an assert instead of trying to fit ```TerminateForDebug()``` into an existing mechanism (i.e. framework events or diagnostic logger)? 
The macro would be a no-op (off) by default and turned on if CppMicroServices is built with a new cmake build option (e.g. ```US_ENABLE_FRAMEWORK_ASSERTS```). 

I'm suggesting this since there are no levels in the diagnostic logger and framework error events don't seem like an appropriate place to terminate the process, even if it is only for debug builds.